### DISCUSSION TITLE
通り芯の作成・表示について

### DISCUSSION THREAD

**Discussion by ShimpeiKojima** [hrntsm/HoaryFox#112] [Ideas]
こんにちは

バージョンのアップデートを準備中ということで、通り芯の作成・表示機能を追加してほしいと思っております。

**Comment by hrntsm** [comment]
今考えている実装は、以下のように通り芯を平面のサーフェスとして表示することを考えていますが、どんな形をイメージしていますか？
![image](https://user-images.githubusercontent.com/23289252/125163308-89b4c880-e1c7-11eb-9d27-4251d9c9d61f.png)
![image](https://user-images.githubusercontent.com/23289252/125163315-90434000-e1c7-11eb-9dea-7acdd1372a82.png)

**Reply by ShimpeiKojima** [reply]
SS7のようにGLで一点鎖線のようなものが表示されているイメージでした！
hironさんご提案のように、srfで表示されるのも良いですね！

**Reply by hrntsm** [reply]
イメージ理解しました。ありがとうございます！
実装コスト考えて線か面かどちらか簡単な方からやってみます。

**Comment by hrntsm** [comment]
試しに実装してみました。こんなイメージでどうでしょうか。  
これまでのコンポーネントと同じようにデータを読み込んだ後、Axisコンポーネントによって軸を作成するようにしています。

![image](https://user-images.githubusercontent.com/23289252/126335487-50820c57-f86b-4fbb-adb8-a16ff7967cb8.png)

**Reply by ShimpeiKojima** [reply]
理想の通りです！ありがとうございます！

**Reply by ShimpeiKojima** [reply]
ちなみに対応バージョンはいくつですか、、、？

**Reply by hrntsm** [reply]
対応は STBv2 なのでそのデータが必要です

**Comment by hrntsm** [comment]
軸線出力機能を付けた gha ファイルをアップしました。使ってみてください。
https://github.com/hrntsm/HoaryFox/releases/tag/v2.0.0-210822

挙動としては以下のような形です。
![image](https://user-images.githubusercontent.com/23289252/130345072-a7c791f6-74f7-4936-963e-0741fd2b99b5.png)
